### PR TITLE
Feat/upgrade monorepo

### PR DESCRIPTION
**Pull request by adamtretera**
- Move modules to monorepo structure
- Added turbo repo 
- Added ts-reset
- T3 env

TODO:
- fix lint issue
- build typeerror
- fix using any useLocalizedResolver 
- add shared scripts (turbo lint, turbo format... ?)

### CODE REVIEW

**Comment by cermakjiri** [comment]
It's definitely a good idea to provide the description with some reasoning why you've added this and that and some overview what has been done, especially with this amount of changes.

**Comment by adamtretera** [comment]
> It's definitely a good idea to provide the description with some reasoning why you've added this and that and some overview what has been done, especially with this amount of changes.

Yeah sorry I actually created a description somehow didnt update the MR

File: template/package.json [hunk 1]
```diff
@@ -1,103 +1,25 @@
...
-            "yarn format"
-        ],
-        "**/*.{png,jpeg,jpg,gif,svg,webp}": [
-            "imagemin-lint-staged"
-        ]
+    "dependencies": {
+        "prettier": "^3.0.3",
+        "turbo": "^1.10.14",
+        "typescript": "^5.2.2",
+        "@manypkg/cli": "^0.21.0"
```
Comment (cermakjiri) [inline template/package.json:19]: It is used somewhere?

File: template/package.json [hunk 2]
```diff
@@ -1,103 +1,25 @@
...
     },
-    "_id": "next-app-template@0.1.0"
-}
\ No newline at end of file
+    "prettier": "@workspace/prettier-config"
```
Comment (cermakjiri) [inline template/package.json]: 👍

File: template/packages/ui/.svgrrc.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+/**
+ * @type {import('@svgr/core').Config}
+ */
+module.exports = {
```
Comment (cermakjiri) [inline template/packages/ui/.svgrrc.js:4]: 👍

File: template/packages/ui/package.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    "license": "MIT",
+    "scripts": {
+        "build": "tsc -p tsconfig.json && tsc-alias",
+        "prepare": "yarn build",
+        "dev": "yarn build && chokidar './src/**/*.{ts,tsx}' -c 'yarn build'",
+        "lint": "eslint \"**/*.{ts,tsx}*\"",
+        "storybook": "storybook dev -p 6006",
+        "build-storybook": "storybook build",
+        "format": "prettier --ignore-path .gitignore \"src/**/*.+(ts|js|tsx|json|html|css|md)\" --write",
+        "generate-icons": "npx @svgr/cli src/components/Icons/Icons/assets && yarn format && eslint --fix \"src/components/Icons/Icons/**/*.tsx\"",
```
Comment (cermakjiri) [inline template/packages/ui/package.json]: I'll remove the npx when it's in the deps.

File: template/tooling/eslint/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+    "extends": "@workspace/tsconfig/base.json",
+    "compilerOptions": {
+        "tsBuildInfoFile": "node_modules/.cache/tsbuildinfo.json"
+    },
+    "include": ["."],
```
Comment (cermakjiri) [inline template/tooling/eslint/tsconfig.json:6]: Shouldn't be the `node_modules/.cache/tsbuildinfo.json` part of `include`?

File: template/tooling/prettier/index.mjs [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+/** @type { PrettierConfig | SortImportsConfig } */
+const config = {
+    ...ackeeConfig,
+    importOrder: [
+        "^(react/(.*)$)|^(react$)|^(react-native(.*)$)",
+        "^(next/(.*)$)|^(next$)",
+        "<THIRD_PARTY_MODULES>",
+        "",
+        "^@workspace/(.*)$",
```
Comment (cermakjiri) [inline template/tooling/prettier/index.mjs]: Nice naming 👍

File: template/tooling/prettier/index.mjs [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import { config as ackeeConfig } from '@ackee/prettier-config';
+
+
+/** @typedef  {import("prettier").Config} PrettierConfig */
+/** @typedef  {import("@ianvs/prettier-plugin-sort-imports").PluginConfig} SortImportsConfig */
```
Comment (cermakjiri) [inline template/tooling/prettier/index.mjs]: `PluginConfig` actually already include PrettierConfig

File: template/tooling/prettier/index.mjs [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+        "^@workspace/(.*)$",
+        "",
+        "^~/",
+        "",
+        "^~/",
+        "^[../]",
+        "^[./]",
+    ],
+};
+export default  config
```
Comment (cermakjiri) [inline template/tooling/prettier/index.mjs]: Looks great, I'll just use `generateImportOrder` from out prettier pkg to simplify it little bit.

File: template/src/modules/logger/index.ts [hunk 9]
```diff
@@ -1,28 +0,0 @@
...
-    captureException(err);
-}
-
-export const logger = {
-    warn,
-    info,
-    debug,
-    trace,
-    error,
-} as const;
```
Comment (cermakjiri) [inline template/src/modules/logger/index.ts]: This has been just deleted. Gonna create pkg for it.

File: template/tooling/prettier/index.mjs (hunk 7 above)
Comment (cermakjiri) [reply template/tooling/prettier/index.mjs]: ✅

File: template/tooling/prettier/index.mjs (hunk 8 above)
Comment (cermakjiri) [reply template/tooling/prettier/index.mjs]: ✅

File: template/src/modules/logger/index.ts (hunk 9 above)
Comment (cermakjiri) [reply template/src/modules/logger/index.ts]: ✅

File: template/packages/ui/package.json (hunk 4 above)
Comment (cermakjiri) [reply template/packages/ui/package.json]: ✅

**Comment by cermakjiri** [comment]
- [x] fix lint issue – ✅ 
- [x] build typeerror – ✅ 
- [x] fix using any useLocalizedResolver – ✅  @2bad2furious  🙌
- [ ] add shared scripts (turbo lint, turbo format... ?) – ⏩  It'd be nice but we need the skeleton ASAP. I suggest to add it later based on the experience from the new projects.
- [ ] Lokse is working 🚧 


Also as we discussed on one the meeting sometime ago, it'd be nice to have `package/common` that includes all shared /  code among apps (`errors`, `localization`, etc.). This pkg doesn't require any build process, it's just plain TS files which are then directly imported to given app. However, I think we can skip for now and added later as well.

**Comment by 2bad2furious** [comment]
Does building the app work now? It failed prerendering home page due to some intl stuff before

**Comment by cermakjiri** [comment]
> Does building the app work now? It failed prerendering home page due to some intl stuff before

That's has been fixed. However lokse doesn't work

File: template/tooling/eslint/tsconfig.json (hunk 5 above)
Comment (adamtretera) [reply template/tooling/eslint/tsconfig.json:6]: No, its not a source code, so it shouldnt

File: template/package.json (hunk 1 above)
Comment (adamtretera) [reply template/package.json:19]: added dependency check script - I think it pretty handy

**Review by cermakjiri — APPROVED** [review]

**Pull request merged**